### ISSUE TITLE
After SAML Login the loading screen stays open

### ISSUE THREAD

**Issue by Czujackt** [RocketChat/Rocket.Chat#23569]
### Description:

We use SAML with Keycloak for login to Rocket.Chat. When you complete the login in the browser, the loading screen stays open. 

### Actual behavior:

![Bildschirmfoto 2021-10-27 um 08 40 35](https://user-images.githubusercontent.com/33861078/139013751-fffaea44-3190-4d7c-adb9-6e90af3a7a60.png)

### Server Setup Information:

- Version of Rocket.Chat Server: 4.0.5
- NodeJS Version: v12.22.1 
- MongoDB Version: 4.4.9

### Client Setup Information

- Desktop App or Browser Version: Safari
- Operating System: macOS

### Relevant logs:

Logs from BROWSER
```
[Error] Error: No callback invoker for method 3
(anonyme Funktion) — aa1d4ab72b86a914f2f370581f059e8299cabbb6.js:772:16473
forEach
_process_updated — aa1d4ab72b86a914f2f370581f059e8299cabbb6.js:772:15890
_processOneDataMessage — aa1d4ab72b86a914f2f370581f059e8299cabbb6.js:772:12800
_livedata_data — aa1d4ab72b86a914f2f370581f059e8299cabbb6.js:772:13389
u — aa1d4ab72b86a914f2f370581f059e8299cabbb6.js:999:275747
(anonyme Funktion) — aa1d4ab72b86a914f2f370581f059e8299cabbb6.js:999:275923
(anonyme Funktion) — aa1d4ab72b86a914f2f370581f059e8299cabbb6.js:1:8268
promiseReactionJob

	(anonyme Funktion) (aa1d4ab72b86a914f2f370581f059e8299cabbb6.js:999:276076)
	(anonyme Funktion) (aa1d4ab72b86a914f2f370581f059e8299cabbb6.js:1:8268)
	promiseReactionJob
```

**Comment by maeries** [comment]
do you have two factor authentication enabled? That caused a similar problem for me. see #23606

**Comment by Czujackt** [comment]
I have tried this, but it makes no difference. My problem is independent of the two factor authentication

**Comment by Nektovet** [comment]
@Czujackt Did you solve the problem? I just caught the same issue

**Comment by Czujackt** [comment]
@Nektovet
No. After constant refreshing of the page it works then eventually.

**Comment by imedia3** [comment]
Same problem, 4.8, 5.0, 5.0.4, 5.1.4, 5.2rc4, with all these versions we have the same issue. 

1) When logging in via SAML, Rocketchat loads blank screen, after refreshing the page (F5) - everything renders. 
2) If you log in via standard login form - the page renders with first load. 
3) No specific logs in Keycloak or Rocketchat. 
4) Google Chrome console logs: 
![image](https://user-images.githubusercontent.com/4366249/194953411-8e21e9df-d055-46b8-aafd-634b2c5f20ea.png)
5) Page's code contains exactly the same code when we see blank page and normal page.

I finished up with a temporary solution by adding following code for authorized users (Settings > Layout > Custom Scripts > Authorized users):
```
if(!window.location.hash) {
       window.location = window.location + '#loaded';
       window.location.reload();
     };
```

**Comment by gabe565** [comment]
> I finished up with a temporary solution by adding following code for authorized users (Settings > Layout > Custom Scripts > Authorized users):

Thanks for the workaround! This worked well for me, but I didn't like adding a hash to the URL. Instead, I added a custom script for logged out users that sets a value in localStorage, and another custom script for logged in users that checks the value, removes it, then refreshes the page.

**Custom Script for Logged Out Users**
```js
localStorage.setItem("will-refresh", "true");
```

**Custom Script for Logged In Users**
```js
if (localStorage.getItem("will-refresh") === "true") {
  localStorage.removeItem("will-refresh");
  window.location.reload();
}
```

**Comment by imedia3** [comment]
> > I finished up with a temporary solution by adding following code for authorized users (Settings > Layout > Custom Scripts > Authorized users):
> 
> Thanks for the workaround! This worked well for me, but I didn't like adding a hash to the URL. Instead, I added a custom script for logged out users that sets a value in localStorage, and another custom script for logged in users that checks the value, removes it, then refreshes the page.
> 
> **Custom Script for Logged Out Users**
> 
> ```js
> localStorage.setItem("will-refresh", "true");
> ```
> 
> **Custom Script for Logged In Users**
> 
> ```js
> if (localStorage.getItem("will-refresh") === "true") {
>   localStorage.removeItem("will-refresh");
>   window.location.reload();
> }
> ```

It is a better solution for my case too, thank for sharing, it is working well
